src/app/plugins/pubsub.plugin.ts

```diff
@@ -1,23 +1,114 @@
+  private _embedSubs = new RichEmbed();
+
+  private _SUB_UPD_THRESH: number = 1000 * 60 * 60 * 24; // in ms, one day.
+  private _LAST_UPD_TIME: number = -this._SUB_UPD_THRESH - 1;
 
   constructor(public container: IContainer) {
     super();
+    this.updateList();
+  }
+
+  private updateList() {
+    this.container.httpService
+      .get(`${this._API_URL}allsubs/`)
+      .then((response: IHttpResponse) => {
+        const subs = response.data;
+
+        this._subs = subs.map((subData: { name: string }) => subData.name.toLowerCase());
+      })
+      .catch((err) => this.container.loggerService.warn(err));
   }
 
   public async execute(message: IMessage, args?: string[]) {
-    const state = this.container.storeService.get('pubSubStore').state;
-    if (state?.onSale) {
-      message.reply('Chicken tender subs are on sale :eyes:');
+    const subIn = this._parseCommand(args || []);
+
+    if (subIn.includes('type') || subIn.includes('list')) {
+      //Simply return the list of supported subs
+      this._generateEmbedSubs();
+      message.reply(this._embedSubs);
+      return;
+    }
+
+    // checks if their sub was a sub, then if that sub is recognised
+    const subType = this._subs.find((sub: string) => sub === subIn) || '';
+
+    if (subType === '' && subIn !== 'random' && subIn !== '') {
+      message.reply('Sub not found.  Use "list" or "type" to get a list of possible subs');
+      return;
+    }
+
+    //recieves the according info and posts, or derps
+    await this.container.httpService
+      .get(`${this._API_URL}subs/?name=${subType}`)
+      .then((response: IHttpResponse) => {
+        const [subData] = response.data;
+        const parsedNamed = subData.sub_name.replace(/-/g, ' ');
+
+        const saleInfo =
+          subData.status.toLowerCase() === 'true' //not sure why I have to do it this way but I do
+            ? `a dicounted price of ${subData.price} until ${subData.last_sale.split('-')[1]}`
+            : `${subData.price}, the last sale Date was ${subData.last_sale}`;
+
+        const embed: RichEmbed = new RichEmbed();
+
+        embed.setColor('#0099ff').setTitle(parsedNamed);
+
+        embed.setDescription(
+          `Get your own *${parsedNamed}* for ${saleInfo}. So what are you waiting for?  Come on down to Publix now to get yourself a beautiful sub. Just look at this beauty right here! 😍😍😍:pubsub::pubsub::pubsub:🤤🤤🤤`
+        );
+
+        embed.setImage(subData.image);
+
+        message.reply(embed);
+      })
+      .catch((err) => this.container.loggerService.warn(err));
+  }
+
+  private async _generateEmbedSubs(): Promise<void> {
+    const lastListUpdate: number = Date.now() - this._LAST_UPD_TIME;
+    if (lastListUpdate < this._SUB_UPD_THRESH) {
+      return;
     } else {
-      message.reply('Chicken tender subs are **NOT** on sale.');
+      this._LAST_UPD_TIME = Date.now(); //since it only updates once a day, don't need to worry about accuracy
     }
```

Comment:
no need for if/else because the if branch just returns. move the else body out  and just have standalone if